rocksdb/tests/test_options.py

```diff
@@ -170,3 +200,436 @@ def test_row_cache(self):
         self.assertIsNone(opts.row_cache)
         opts.row_cache = cache = rocksdb.LRUCache(2*1024*1024)
         self.assertEqual(cache, opts.row_cache)
+
+    def test_max_open_files(self):
+        opts = rocksdb.Options()
+        self.assertIsNotNone(opts.max_open_files)
+        opts.max_open_files = 10
+        self.assertEqual(opts.max_open_files, 10)
```

Comment:
These tests are all almost the same code, you could do this in a much more compact way:
```suggestion
    def test_rockdb_options(self):
        NOTNONE = object()
        UNSETTABLE = object()
        for option, def_value, new_value in (
            ('max_open_files', NOTNONE, 10),
            ('row_cache', None, UNSETTABLE),
            ...,
        ):
            with self.subTest(option=option):
                opts = rocksdb.Options()
                if def_value is NOTNONE:
                    self.assertIsNotNone(getattr(opts, option))
                else:
                    self.assertEqual(def_value, getattr(opts, option))
                if new_value is UNSETTABLE:
                    self.assertRaises(
                        Exception, setattr, opts, option, new_value)
                else:
                    setattr(opts, option, new_value)
                    self.assertEqual(getattr(opts, option), new_value)
```

Reply:
This helped a lot. Thanks!